### PR TITLE
Add IsNotWindowsNanoServer to Platform detection.

### PR DESCRIPTION
**Pull request by Priya91**
Disable failing Windows Nano tests.

### CODE REVIEW

File: src/System.Diagnostics.Process/tests/ProcessStandardConsoleTests.cs [hunk 1]
```diff
@@ -12,7 +12,7 @@ public class ProcessStandardConsoleTests : ProcessTestBase
     {
         private const int s_ConsoleEncoding = 437;
 
-        [Fact]
+        [ConditionalFact(nameof(PlatformDetection)+"."+nameof(PlatformDetection.IsNotWindowsNanoServer))]
```
Comment (mellinoe) [inline src/System.Diagnostics.Process/tests/ProcessStandardConsoleTests.cs]: nit: Formatting is different from below attribute (spaces between operands). Consider using interpolated strings (might not look very good though)

**Comment by mellinoe** [comment]
LGTM

**Comment by Priya91** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test

File: src/System.Diagnostics.Process/tests/ProcessStandardConsoleTests.cs (hunk 1 above)
Comment (Priya91) [inline src/System.Diagnostics.Process/tests/ProcessStandardConsoleTests.cs]: VS must have auto formatted the other files when opened. Interpolated strings wont work here, as the attribute should have constant values as arguments, it cannot be expressions..

Comment (mellinoe) [inline src/System.Diagnostics.Process/tests/ProcessStandardConsoleTests.cs]: Ah yes, good point about interpolated strings.

**Comment by Priya91** [comment]
Merging, since the last commit is only on formatting.

File: src/Common/tests/System/PlatformDetection.cs [hunk 2]
```diff
@@ -13,9 +13,11 @@ namespace System
     public static class PlatformDetection
     {
         public static bool IsWindows { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Windows);
-        public static bool IsWindows7 { get; } =  IsWindows && GetWindowsVersion() == 6 && GetWindowsMinorVersion() == 1;
+        public static bool IsWindows7 { get; } = IsWindows && GetWindowsVersion() == 6 && GetWindowsMinorVersion() == 1;
         public static bool IsOSX { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
         public static bool IsNetBSD { get; } = RuntimeInformation.IsOSPlatform(OSPlatform.Create("NETBSD"));
+        public static bool IsNotWindowsNanoServer { get; } = (IsWindows && GetWindowsVersion() == 10 &&
```
Comment (jkotas) [inline src/Common/tests/System/PlatformDetection.cs]: Does this property return `false` for Win7 and Win8? It means that the affected tests will be disabled for nanoserver, but also for Win7 and Win8...

I think the implementation was meant to be:
```
public static bool IsNotWindowsNanoServer { get; } = (IsWindows && (GetWindowsVersion() < 10 || File.Exists(...)));
```

Comment (Priya91) [inline src/Common/tests/System/PlatformDetection.cs]: Yeah, that's correct :) Thanks for checking!

**Pull request merged**